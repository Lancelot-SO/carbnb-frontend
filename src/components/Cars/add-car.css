/* .form-container {
  display: flex;
  flex-direction: column;
  padding: 0.5rem;
  align-items: center;
  justify-content: center;
  width: 75%;
}

.form-container label {
  padding: 0.8rem 1rem 0;
  color: #9ca3af;
}

form {
  display: flex;
  flex-direction: column;
  width: 100%;
  margin-top: -3rem;
}

input[type="text"],
input[type="integer"],
input[type="url"],
input[type="number"],
input[type="username"],
select,
input[type="date"] {
  padding: 1rem;
  border: none;
  border-bottom: 1px solid #acacac;
  width: 65%;
} */

.form-container {
  max-width: 400px;
  margin: 0 auto;
}

.form-container > h2 {
  color: green;
  margin-top: 20px;
  text-align: center;
}

.form-container form {
  display: flex;
  flex-direction: column;
}

.form-container input {
  margin-bottom: 10px;
  padding: 10px;
  border: 1px solid #ccc;
  border-radius: 4px;
}

.form-container label {
  margin-bottom: 5px;
}

.form-container .btn-group {
  margin-top: 10px;
  display: flex;
  justify-content: center;
}

.form-container button {
  padding: 10px 20px;
  background-color: #4caf50;
  color: #fff;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}

.form-container button:hover {
  background-color: #45a049;
}
