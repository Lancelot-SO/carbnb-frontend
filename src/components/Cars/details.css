
.details-container {
  width: 80%;
  height: 75vh;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 2rem;
}

.details-image {
  width: 65%;
  border-radius: 2rem;
  margin: 1rem;
}

.details-info {
  margin: 2rem;
  text-align: right;
  width: 25%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.details-name {
  font-size: 32px;
  font-weight: 600;
  text-transform: uppercase;
  padding-right: 10px;
}

.details-description {
  padding-right: 10px;
}

.details-list {
  display: flex;
  flex-direction: column;
  gap: 5px;
  padding: 10px;
  text-align: left;
  width: 100%;
}

.details-list li {
  background-color: white;
  padding: 10px;
  display: flex;
  justify-content: space-between;
}

.details-list li:nth-child(even) {
  background-color: #e0e0e0;
}

.details-link-container {
  padding-top: 3rem;
}

.details-link {
  text-transform: uppercase;
  font-size: 12px;
  font-weight: 600;
  letter-spacing: -0.02em;
  margin-right: 15px;
}

.reserve-btn-container {
  display: flex;
  align-items: flex-end;
  justify-content: flex-end;
}

.details-reserve-btn {
  color: #fff;
  font-size: 19px;
  font-weight: 600;
  background-color: var(--color-secondary);
  width: 175px;
  border-radius: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 15px;
}

.details-reserve-btn > a {
  display: flex;
  flex-direction: row;
  gap: 25px;
  text-align: center;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  padding: 18px 25px;
}

.details-btn {
  display: flex;
  flex-direction: row;
  gap: 20px;
}

.reserve-text {
  font-size: 20px;
  color: #fff;
  font-weight: 500;
}

@media only screen and (max-width: 980px) {
  .details-container {
    flex-direction: column;
    justify-content: center;
    align-items: baseline;
    height: auto;
    max-height: 100%;
    margin-right: 2rem;
  }

  .details-image {
    height: 70vh;
    width: 100%;
  }

  .details-info {
    width: 100%;
  }

  .details-link-container {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }

  .details-reserve-btn > a {
    padding: 15px 20px;
  }
}
