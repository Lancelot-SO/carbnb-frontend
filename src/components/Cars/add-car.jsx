import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import './add-car.css';
import { addCar } from '../../redux/Actions/car-actions';
import { fetchUsers } from '../../redux/Actions/user-actions';

const AddCar = () => {
  const dispatch = useDispatch();

  const [name, setName] = useState('');
  const [user, setUser] = useState(null);

  useEffect(() => {
    dispatch(fetchUsers());
  }, [dispatch]);

  const users = useSelector((state) => state.usersSlice.users[0]);

  useEffect(() => {
    setUser(users?.find((user) => user.username === JSON.parse(localStorage.getItem('user'))) || 0);
  }, [users]);

  console.log(user);

  const [description, setDescription] = useState('');
  const [imageUrl, setImageUrl] = useState('');
  const [price, setPrice] = useState(0);
  const [model, setModel] = useState('');

  const createNewCar = (user, name, description, imageUrl, price, model) => ({
    user,
    name,
    description,
    imageUrl,
    price,
    model,
  });

  const handleSubmit = (e) => {
    e.preventDefault();
    const car = createNewCar(
      user.id,
      name,
      description,
      imageUrl,
      price,
      model,
    );
    dispatch(addCar(car));
    setName('');
    setDescription('');
    setImageUrl('');
    setPrice(0);
    setModel('');
  };

  return (
    <div className="form-container">
      <h2>Add Car</h2>
      <form onSubmit={handleSubmit}>
        <input placeholder="Name" type="text" name="Name" value={name} onChange={(e) => setName(e.target.value)} required />
        <input placeholder="Description" type="text" name="Description" value={description} onChange={(e) => setDescription(e.target.value)} required />
        <input placeholder="Image URL" type="url" name="ImageUrl" value={imageUrl} onChange={(e) => setImageUrl(e.target.value)} required />
        <input id="model" placeholder="Model" type="text" name="Model" value={model} onChange={(e) => setModel(e.target.value)} required />
        <label htmlFor="price">Price</label>
        <input id="price" placeholder="Price" type="number" name="Price" value={price} onChange={(e) => setPrice(e.target.value)} required />
        <div className="btn-group">
          <button type="submit">Add Car</button>
        </div>
      </form>
    </div>
  );
};

export default AddCar;
