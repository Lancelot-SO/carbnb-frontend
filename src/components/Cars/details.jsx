import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useLocation, Link } from 'react-router-dom';
import { fetchUserCars } from '../../redux/Actions/car-actions';
import { resetState } from '../../redux/Slices/carDetails-slice';
import './details.css';

export const Details = () => {
  const dispatch = useDispatch();
  const location = useLocation();

  const id = location.state?.id;

  useEffect(() => {
    dispatch(resetState());
  }, [dispatch]);

  useEffect(() => {
    dispatch(fetchUserCars(id));
  }, [dispatch, id]);

  const details = useSelector((state) => state.carDetailsSlice[0]?.car);

  const imageSectionStyles = {
    backgroundImage: `url(${details?.image})`,
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
  };

  return (
    <div className="details-container">
      <section style={imageSectionStyles} className="details-image" />
      <section className="details-info">
        <div className="details-top">
          <h5 className="details-name">{details?.name}</h5>
          <p className="details-description">{details?.description}</p>
          <ul className="details-list">
            <li>
              <span>Model</span>
              <span>{details?.model}</span>
            </li>
            <li>
              <span>Price</span>
              <span>{details?.price}</span>
            </li>
          </ul>
          <div className="details-link-container">
            <Link className="details-link" to="/">
              discover more models
              <span>âž¤</span>
            </Link>
          </div>
        </div>
        <div className="reserve-btn-container">
          <div className="details-reserve-btn">
            <Link to="/reserve" state={details}>
              <button type="button" className="details-btn">
                <span className="reserve-text">Reserve</span>
                <img className="right-icon" alt="reserve-icon" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAYAAAA7MK6iAAAACXBIWXMAAAsTAAALEwEAmpwYAAACAklEQVR4nM3Xu24TURAGYCM5OHSEwpQQCgoQFQ0NJHFoIMCbgAS8AV0gOI8QjAhIRDQQLooicZHzDkAEb0ACgiJUHzrKFBuwvbu+iBzpSEezc+bfmZ35Z7ZS2e8Lh3AVC1jG+9jpfA9XMD5MwBN4iF/4hqe4j5ux03kFW6HzAJODANbQxG88xwyqPfSraGA17qTI1MqCHsUGPuJ8Hy89hc9oo14G9Cve4HBZ0IydCazhSy54hHcjQLuGtQR4NcDbONhLsRnh7dvTLp5/SpnfK3tTUlzIMXQaN0qCT2OnY7ZHybwoYOQMvmOxJPhLtDqRQ6rBmYJGzpUFxyx+7iEZXAtyKJxQOBuksYgDBfTHQn8uK0yUt1IUtF/P8Qx3s4InXbNuiJ7bpdflrOAtbvUDnPH8B+7k6N1OWPsC+HEi9UFDXUC3iUf/K7nms4LUxLdTyo/C09AfC4xLWeF4FHdjhARy8R8CiQet1MQLGDjVJ2W+wlKnB5NB5FM5Bk7ieknQRtg+3k1hIVrYRBnDOaBHsLknqboMAu1o3sMYBFJCreNDz0EglOsxrqwN4nl4ul5o9PkLvB1hn66UXPFNN8PTetnLtSCWnWjis73qPMKaSuZ13JnPDW+vFdneikFhK9inmRnomyHbjjpdwrHKsJZdkplL/TTxLd7FTuckuzzUX5hRrT+dIUr6Buw0mQAAAABJRU5ErkJggg==" />
              </button>
            </Link>
          </div>
        </div>
      </section>
    </div>
  );
};

export default Details;
