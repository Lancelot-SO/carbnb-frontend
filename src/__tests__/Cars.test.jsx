import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { describe, it, expect } from 'vitest';
import '@testing-library/jest-dom';
import store from '../redux/index';
import Cars from '../components/Cars/details';

describe('App', () => {
  it('renders Cars', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Cars />
        </BrowserRouter>
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
