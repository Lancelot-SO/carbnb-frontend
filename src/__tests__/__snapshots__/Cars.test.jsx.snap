// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders Cars 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="details-container"
      >
        <section
          class="details-image"
          style="background-image: url(undefined); background-size: cover; background-repeat: no-repeat; background-position: center;"
        />
        <section
          class="details-info"
        >
          <div
            class="details-top"
          >
            <h5
              class="details-name"
            />
            <p
              class="details-description"
            />
            <ul
              class="details-list"
            >
              <li>
                <span>
                  Model
                </span>
                <span />
              </li>
              <li>
                <span>
                  Price
                </span>
                <span />
              </li>
            </ul>
            <div
              class="details-link-container"
            >
              <a
                class="details-link"
                href="/"
              >
                discover more models
                <span>
                  ➤
                </span>
              </a>
            </div>
          </div>
          <div
            class="reserve-btn-container"
          >
            <div
              class="details-reserve-btn"
            >
              <a
                href="/reserve"
              >
                <button
                  class="details-btn"
                  type="button"
                >
                  <span
                    class="reserve-text"
                  >
                    Reserve
                  </span>
                  <img
                    alt="reserve-icon"
                    class="right-icon"
                    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAYAAAA7MK6iAAAACXBIWXMAAAsTAAALEwEAmpwYAAACAklEQVR4nM3Xu24TURAGYCM5OHSEwpQQCgoQFQ0NJHFoIMCbgAS8AV0gOI8QjAhIRDQQLooicZHzDkAEb0ACgiJUHzrKFBuwvbu+iBzpSEezc+bfmZ35Z7ZS2e8Lh3AVC1jG+9jpfA9XMD5MwBN4iF/4hqe4j5ux03kFW6HzAJODANbQxG88xwyqPfSraGA17qTI1MqCHsUGPuJ8Hy89hc9oo14G9Cve4HBZ0IydCazhSy54hHcjQLuGtQR4NcDbONhLsRnh7dvTLp5/SpnfK3tTUlzIMXQaN0qCT2OnY7ZHybwoYOQMvmOxJPhLtDqRQ6rBmYJGzpUFxyx+7iEZXAtyKJxQOBuksYgDBfTHQn8uK0yUt1IUtF/P8Qx3s4InXbNuiJ7bpdflrOAtbvUDnPH8B+7k6N1OWPsC+HEi9UFDXUC3iUf/K7nms4LUxLdTyo/C09AfC4xLWeF4FHdjhARy8R8CiQet1MQLGDjVJ2W+wlKnB5NB5FM5Bk7ieknQRtg+3k1hIVrYRBnDOaBHsLknqboMAu1o3sMYBFJCreNDz0EglOsxrqwN4nl4ul5o9PkLvB1hn66UXPFNN8PTetnLtSCWnWjis73qPMKaSuZ13JnPDW+vFdneikFhK9inmRnomyHbjjpdwrHKsJZdkplL/TTxLd7FTuckuzzUX5hRrT+dIUr6Buw0mQAAAABJRU5ErkJggg=="
                  />
                </button>
              </a>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="details-container"
    >
      <section
        class="details-image"
        style="background-image: url(undefined); background-size: cover; background-repeat: no-repeat; background-position: center;"
      />
      <section
        class="details-info"
      >
        <div
          class="details-top"
        >
          <h5
            class="details-name"
          />
          <p
            class="details-description"
          />
          <ul
            class="details-list"
          >
            <li>
              <span>
                Model
              </span>
              <span />
            </li>
            <li>
              <span>
                Price
              </span>
              <span />
            </li>
          </ul>
          <div
            class="details-link-container"
          >
            <a
              class="details-link"
              href="/"
            >
              discover more models
              <span>
                ➤
              </span>
            </a>
          </div>
        </div>
        <div
          class="reserve-btn-container"
        >
          <div
            class="details-reserve-btn"
          >
            <a
              href="/reserve"
            >
              <button
                class="details-btn"
                type="button"
              >
                <span
                  class="reserve-text"
                >
                  Reserve
                </span>
                <img
                  alt="reserve-icon"
                  class="right-icon"
                  src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAYAAAA7MK6iAAAACXBIWXMAAAsTAAALEwEAmpwYAAACAklEQVR4nM3Xu24TURAGYCM5OHSEwpQQCgoQFQ0NJHFoIMCbgAS8AV0gOI8QjAhIRDQQLooicZHzDkAEb0ACgiJUHzrKFBuwvbu+iBzpSEezc+bfmZ35Z7ZS2e8Lh3AVC1jG+9jpfA9XMD5MwBN4iF/4hqe4j5ux03kFW6HzAJODANbQxG88xwyqPfSraGA17qTI1MqCHsUGPuJ8Hy89hc9oo14G9Cve4HBZ0IydCazhSy54hHcjQLuGtQR4NcDbONhLsRnh7dvTLp5/SpnfK3tTUlzIMXQaN0qCT2OnY7ZHybwoYOQMvmOxJPhLtDqRQ6rBmYJGzpUFxyx+7iEZXAtyKJxQOBuksYgDBfTHQn8uK0yUt1IUtF/P8Qx3s4InXbNuiJ7bpdflrOAtbvUDnPH8B+7k6N1OWPsC+HEi9UFDXUC3iUf/K7nms4LUxLdTyo/C09AfC4xLWeF4FHdjhARy8R8CiQet1MQLGDjVJ2W+wlKnB5NB5FM5Bk7ieknQRtg+3k1hIVrYRBnDOaBHsLknqboMAu1o3sMYBFJCreNDz0EglOsxrqwN4nl4ul5o9PkLvB1hn66UXPFNN8PTetnLtSCWnWjis73qPMKaSuZ13JnPDW+vFdneikFhK9inmRnomyHbjjpdwrHKsJZdkplL/TTxLd7FTuckuzzUX5hRrT+dIUr6Buw0mQAAAABJRU5ErkJggg=="
                />
              </button>
            </a>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
